site/app/models/gradeable/LateDayInfo.php

```diff
@@ -71,6 +100,51 @@ public function toArray() {
         ];
     }
 
+    /**
+     * Gets information about this late day event in the format for the cache table
+     * @return array
+     */
+    public function generateEventInfo() {
```

Comment:
Same comment here about type-hinting (applies to all of the following functions)